src/main/java/com/aws/greengrass/secretmanager/SecretManagerService.java

```diff
@@ -257,71 +251,18 @@
         } catch (Throwable t) {
             status = 500;
             message = t.getMessage();
-            logger.atError().event("secret-access").setCause(t)
-                    .kv("Principal", serviceName).kv("secret", new String(request))
-                    .log("Error getting secret");
+            logger.atError().event("secret-access").setCause(t).kv("Principal", serviceName)
+                    .kv("secret", new String(request)).log("Error getting secret");
         }
 
         try {
-            return OBJECT_MAPPER.writeValueAsBytes(GetSecretValueError.builder().status(status).message(message)
-                    .build());
+            return OBJECT_MAPPER.writeValueAsBytes(
+                    GetSecretValueError.builder().status(status).message(message).build());
         } catch (JsonProcessingException e) {
-            logger.atError().event("secret-access").setCause(e)
-                    .kv("Principal", serviceName)
-                    .kv("secret", new String(request))
-                    .kv("status", status)
-                    .kv("message", message)
+            logger.atError().event("secret-access").setCause(e).kv("Principal", serviceName)
+                    .kv("secret", new String(request)).kv("status", status).kv("message", message)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### new String [](|8df4fb67-a24a-4a26-a18a-395ef71250d0|)
Java Strings use Unicode internally to encode characters. 

When constructing a string from a `byte[]`, a character encoding must be used to convert the bytes to a String&#39;s internal encoding. When no character encoding is passed explicitly, the Java runtime will use a default encoding. This encoding usually isn&#39;t UTF-8 and might become misconfigured. When converting bytes to characters or strings, always supply the actual character encoding to use.

#### Suggested solutions:
Specify an encoding

*(likely `UTF-8`, which is backward compatible with ASCII and has multi-language support)*

```
String message = new String(messageBytes, StandardCharsets.UTF_8);
```